charms/worker/k8s/src/token_distributor.py

```diff
@@ -245,12 +269,20 @@ def allocate_tokens(
                     relation.name,
                     unit.name,
                 )
-                app_databag[unit.name] = name
-                if invalidate_on_join:
-                    relation.data[self.charm.unit].pop(secret_id, None)
+                app_databag[unit.name] = f"joined-{name}"
+                if revoke_on_join:
+                    self._revoke_juju_secret(relation, unit)
+
                 continue  # unit reports its joined already
             if relation.data[self.charm.unit].get(secret_id):
-                continue  # unit already assigned a token
+                # unit already assigned a token
+                log.info(
+                    "Waiting for token to be recovered %s unit=%s:%s",
+                    token_type.value,
+                    relation.name,
+                    unit.name,
+                )
+                continue
```

Comment:
I wanted a log here to see if the leader control plane was ever seeing it had handed out a token, that hadn't yet been used